Source/Diagnostics/ReducedDiags/FieldProbe.cpp

```diff
@@ -147,102 +189,219 @@ void FieldProbe::ComputeDiags (int step)
+                // Interpolating to the probe positions for each particle
+
+                // Temporarily defining modes and interp outside ParallelFor to avoid GPU compilation errors.
+                const int temp_modes = WarpX::n_rz_azimuthal_modes;
+                const int temp_interp_order = interp_order;
+                const bool temp_raw_fields = raw_fields;
+                const bool temp_field_probe_integrate = m_field_probe_integrate;
+
+                long const np = pti.numParticles();
+                amrex::ParallelFor( np, [=] AMREX_GPU_DEVICE (long ip)
+                {
+                    amrex::ParticleReal xp, yp, zp;
+                    getPosition(ip, xp, yp, zp);
+
+                    amrex::ParticleReal Exp = 0._rt, Eyp = 0._rt, Ezp = 0._rt;
+                    amrex::ParticleReal Bxp = 0._rt, Byp = 0._rt, Bzp = 0._rt;
+
+                    // first gather E and B to the particle positions
+                    doGatherShapeN(xp, yp, zp, Exp, Eyp, Ezp, Bxp, Byp, Bzp,
+                                   arrEx, arrEy, arrEz, arrBx, arrBy, arrBz,
+                                   Extype, Eytype, Eztype, Bxtype, Bytype, Bztype,
+                                   dx_arr, xyzmin_arr, amrex::lbound(box), temp_modes,
+                                   temp_interp_order, false);
+
+                    //Calculate the Poynting Vector S
+                    amrex::Real const sraw[3]{
+                    Exp * Bzp - Ezp * Byp,
+                    Ezp * Bxp - Exp * Bzp,
+                    Exp * Byp - Eyp * Bxp};
+                    amrex::ParticleReal const S = (1._rt / PhysConst::mu0)  * sqrt(sraw[0] * sraw[0] + sraw[1] * sraw[1] + sraw[2] * sraw[2]);
+
+                    /*
+                     * Determine whether or not to integrate field data.
+                     * If not integrating, store instantaneous values.
+                     */
+                    if (temp_field_probe_integrate)
+                    {
+
+                        // store values on particles
+                        part_Ex[ip] += Exp; //remember to add lorentz transform
+                        part_Ey[ip] += Eyp; //remember to add lorentz transform
+                        part_Ez[ip] += Ezp; //remember to add lorentz transform
+                        part_Bx[ip] += Bxp; //remember to add lorentz transform
+                        part_By[ip] += Byp; //remember to add lorentz transform
+                        part_Bz[ip] += Bzp; //remember to add lorentz transform
+                        part_S[ip] += S; //remember to add lorentz transform
+                    }
+                    else
+                    {
+                        // Either save the interpolated fields or the raw fields depending on the raw_fields flag
+                        part_Ex[ip] = temp_raw_fields ? arrEx(i_probe, j_probe, k_probe) : Exp; //remember to add lorentz transform
```

Comment:
We should move this logic up, see comment above.

Logic here will then read:
```suggestion
                        part_Ex[ip] = Exp; //remember to add lorentz transform
```